### ISSUE TITLE
Misfire MD025

### ISSUE THREAD

**Issue by unional** [DavidAnson/vscode-markdownlint#10]
```markdown
- something
- # of other things  <-- MD025
- something
```

**Comment by DavidAnson** [comment]
Using https://dlaa.me/markdownlint/ as a test environment, your sample does not report MD025 "Multiple top level headers in the same document" for me, but this does:

```md
# heading

- something
- # of other things  <-- MD025
- something
```

That's because headers are allowed in lists, so there are really two top-level headers in the example I show. Here's a link to the spec I used to verify that: http://spec.commonmark.org/0.27/#example-261

**Comment by DavidAnson** [comment]
If you need to escape the list item so it's not a header, add a backslash like this:

```md
- \# of other things  <-- MD025
```

**Comment by DavidAnson** [comment]
Please let me know if the above does not resolve your issue.

**Comment by unional** [comment]
yes, that solves it. Why list item support headers?

**Comment by DavidAnson** [comment]
I'm not sure - but it's in the specification. :)
